openmdao/recorders/sqlite_recorder.py

```diff
@@ -594,13 +594,14 @@ def record_metadata_system(self, recording_requester):
             try:
                 pickled_metadata = pickle.dumps(user_options, self._pickle_version)
             except Exception:
-                pickled_metadata = pickle.dumps(OptionsDictionary(), self._pickle_version)
-                simple_warning("Trying to record options which cannot be pickled "
-                               "on system with name: %s. Use the 'options_excludes' "
-                               "recording option on system objects to avoid attempting "
-                               "to record options which cannot be pickled. Skipping "
-                               "recording options for this system." % recording_requester.name,
-                               RuntimeWarning)
+                try:
+                    for key, values in user_options._dict.items():
```

Comment:
Wouldn't it be more efficient to do the full pickle first, then if there is an exception, in the except block, loop over each one to find the name of the one that fails.  Otherwise, we are pickling everything twice the majority of the time.

Reply:
That's what we're doing already

Reply:
you are right, I didn't notice we were already inside an exception block.